syntaxes/test/data/expression.html

```diff
@@ -0,0 +1,63 @@
+<!-- Interpolation test -->
+<div>{{ true ? call(1 + 2 + 3) : 'test'  }}</div>
+<div>{{ testing.call(1 + 2 + 3, another?.one()).test }}</div>
+
+<td mat-cell
+    *ngFor="let param of params | async as p; last as last; first as first">
+    {{ param.costCode?.name(test, {test: 'test'} 'test', 123) }}
+</td>
+
+<!-- Property binding test -->
+<div [ ngStyle ]="{
+  'max-width.px': i * 2 + 5
+}"></div>
+<div [ngClass]="{
+    'class-one': test.test(),
+    'class-two',
+    'class-three'
+}"></div>>
+
+<div [@animation.trigger]="val"></div>
+<img [attr.aria-label]="val" />
+<div [my-property]="val"></div>
+<div [my_property]="val"></div>
+<div [myProperty$]="val"></div>
+<div [%invalidProperty]="val"></div>
+<div [invalidProperty)="val"></div>
+
+<!-- Event binding test -->
```

Comment:
Can you refactor this test file to more appropriately test the various scopes provided by your grammar? The expression grammar test probably doesn't care too much that the event binding is picked up probably, because that is covered by the template grammar test. Here we should test that the different strings, expression operators, etc. work correctly.

Reply:
Sure. I'll do it tonight